src/helpers.ml

```diff
@@ -86,11 +86,16 @@ let unmarshal_int32 s = Int64.to_int32 (unmarshal_int 4 s)
 let unmarshal_int64 s = unmarshal_int 8 s
 
 let unmask mask str =
-  if String.length str > 0 then begin
+  if String.length str = 0 then str
+  else
+    let buf = Bytes.of_string str in
```

Comment:
I agree with this choice here, even though we have to make a copy, the docs warn against mutating the output of `unsafe_of_string`, that could result in bugs: https://caml.inria.fr/pub/docs/manual-ocaml/libref/Bytes.html#VALunsafe_of_string

Reply:
However, this does result in a copy, and the question is how performance-sensitive this code is.
I see the call flow is like `<iterator magic> -> readn --> unmask`, we could try changing the `Chunk of string` type to `Chunk of bytes` if this makes sense and this code is performance-sensitive. But hopefully performance doesn't matter too much here and in that case this is good to merge.

Reply:
I don't know but I think the masked chunks are generally small enough

Reply:
You can see how the code would look like otherwise here: https://github.com/mseri/wsproxy/commit/6fae39d4d4a332fc75a74207a4a34a30cd870b85

I feel much more confident reducing the amount and scope of the changes and having this copy to be honest.

Reply:
Don't you set all the elements of the buffer in the loop below? In that case can't the buf be initialized to all 0 or even uninitialized?

Reply:
Yes, it's true. I didn't think of making it uninitialised. That would make the code do just one traversal of the string, like the old one!